### PR TITLE
Hearing - Removed `remoteExec` & added microoptimisations

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- Title.
- Moved `TRACE` down to where it's actually called, in case it's called before CBA settings are ready.

@BrettMayson you added `remoteExec`, is there any specific reason why? I couldn't find anything on the PR.

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/hearing/XEH_preInit.sqf [hunk 1]
```diff
@@ -8,11 +8,14 @@ PREP_RECOMPILE_END;
...
 
+[QGVAR(updateVolume), LINKFUNC(updateVolume)] call CBA_fnc_addEventHandler;
+
 ["CBA_loadoutSet", {
     params ["_unit", "_loadout", "_extendedInfo"];
     if (_extendedInfo getOrDefault ["ace_earplugs", false]) then {
         _unit setVariable ["ACE_hasEarPlugsIn", true, true];
-        [[true]] remoteExec [QFUNC(updateVolume), _unit];
+
+        [QGVAR(updateVolume), [true], _unit] call CBA_fnc_targetEvent;
```
Comment (PabstMirror) [inline addons/hearing/XEH_preInit.sqf:18]:
```suggestion
        [QGVAR(updateVolume), [[true]], _unit] call CBA_fnc_targetEvent;
```

Comment (johnb432) [reply addons/hearing/XEH_preInit.sqf:18]: I tested both scenarios, it doesn't seem to make a difference. However, it does make more sense to update it.

**Comment by PabstMirror** [comment]
I think it was possible for `updateVolume` to run on dedicated servers
so I moved the EH to after the `!hasInterface` and setting check

**Comment by johnb432** [comment]
> I think it was possible for `updateVolume` to run on dedicated servers so I moved the EH to after the `!hasInterface` and setting check

I added it in preInit, because the CBA loadout events are registered in preInit.

**Pull request merged**